src/Servers/Kestrel/shared/TransportConnection.FeatureCollection.cs

```diff
@@ -41,6 +43,8 @@ CancellationToken IConnectionLifetimeFeature.ConnectionClosed
             set => ConnectionClosed = value;
         }
 
+        Socket? IConnectionSocketFeature.Socket { get; }
```

Comment:
This should look like:

```
Socket? IConnectionSocketFeature.Socket => Socket;
```

That would allow the derived class to specify the implementation.

Reply:
There should be a virtual Socket property in here https://github.com/dotnet/aspnetcore/blob/e7b6a959925da2b29bb753ecd379adaf1cbb5dcd/src/Servers/Kestrel/shared/TransportConnection.cs

Reply:
Thanks, @davidfowl. I already had those added in first my commit, but then you asked me to remove them, so I removed them! 
But I will get those added in my latest commit! 

Thanks again!